proxy_agent/src/shared_state.rs

```diff
@@ -108,13 +117,73 @@ impl Default for SharedState {
             telemetry_reader_shutdown: false,
             telemetry_logger_shutdown: false,
             telemetry_logger_status_message: UNKNOWN_STATUS_MESSAGE.to_string(),
+            mock_server_ip: None,   // test only
+            mock_server_port: None, // test only
             // service
             #[cfg(windows)]
             service_status_handle: None,
         }
     }
 }
 
+pub mod shared_state_wrapper {
```

Comment:
NIT: I think this module is better named `shared_state_helpers`, as a lot of the functions are helpers for properly working on the shared state object or accessing things from it.